lib/Doctrine/DBAL/Migrations/Tools/Console/Command/MigrateCommand.php

```diff
@@ -124,31 +124,37 @@ public function execute(InputInterface $input, OutputInterface $output)
         if ($path = $input->getOption('write-sql')) {
             $path = is_bool($path) ? getcwd() : $path;
             $migration->writeSqlFile($path, $version);
-        } else {
-            $dryRun = (boolean) $input->getOption('dry-run');
-
-            // warn the user if no dry run and interaction is on
-            if (! $dryRun) {
-                $question = 'WARNING! You are about to execute a database migration'
-                    . ' that could result in schema changes and data lost.'
-                    . ' Are you sure you wish to continue? (y/n)';
-                if (! $this->canExecute($question, $input, $output)) {
-                    $output->writeln('<error>Migration cancelled!</error>');
-
-                    return 1;
-                }
-            }
+            return 0;
+        }
 
-            $migration->setNoMigrationException($input->getOption('allow-no-migration'));
-            $sql = $migration->migrate($version, $dryRun, $timeAllqueries);
+        $dryRun = (boolean) $input->getOption('dry-run');
 
-            if (empty($sql)) {
-                $output->writeln('<comment>No migrations to execute.</comment>');
-            }
+        $migration->setNoMigrationException($input->getOption('allow-no-migration'));
+        $result = $migration->migrate($version, $dryRun, $timeAllqueries, function () use ($input, $output) {
+            $question = 'WARNING! You are about to execute a database migration'
+                . ' that could result in schema changes and data lost.'
+                . ' Are you sure you wish to continue? (y/n)';
+            return $this->canExecute($question, $input, $output);
+        });
+
+        if (false === $result) {
+            $output->writeln('<error>Migration cancelled!</error>');
+            return 1;
         }
     }
 
     /**
+     * Create a new migration instance to execute the migrations.
+     *
+     * @param $configuration The configuration with which the migrations will be executed
+     * @return Migration a new migration instance
+     */
+    protected function createMigration(Configuration $configuration)
```

Comment:
Purely to make the command easier to unit test using [`getMockForAbstractClass` to override this factory method](https://github.com/chrisguitarguy/migrations/blob/b9f207b9423d27f7e2c9e192f24b43d0817edb89/tests/Doctrine/DBAL/Migrations/Tests/Tools/Console/Command/MigrateCommandTest.php#L193-L205). Ideally there'd probably be a factory object/interface or something, but that seemed like overkill.

Some of the functional tests cover the happy paths that deal with an actual `Migration` object, but the unit test coverage of the command was pretty slim.